samples/dnn/scene_text_spotting.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <fstream>
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/dnn/dnn.hpp>
+
+using namespace cv;
+using namespace cv::dnn;
+
+std::string keys =
+        "{ help  h                          | | Print help message. }"
+        "{ inputImage i                     | | Path to an input image. Skip this argument to capture frames from a camera. }"
+        "{ detModelPath dmp                 | | Path to a binary .onnx model for detection. "
+            "Download links are provided in doc/tutorials/dnn/dnn_text_spotting/dnn_text_spotting.markdown}"
+        "{ recModelPath rmp                 | | Path to a binary .onnx model for recognition. "
+            "Download links are provided in doc/tutorials/dnn/dnn_text_spotting/dnn_text_spotting.markdown}"
+        "{ inputHeight ih                   |736| image height of the model input. }"
+        "{ inputWidth iw                    |1280| image width of the model input. }"
+        "{ RGBInput rgb                     |0| 0: imread with flags=IMREAD_GRAYSCALE; 1: imread with flags=IMREAD_COLOR. }"
+        "{ binaryThreshold bt               |0.3| Confidence threshold of the binary map. }"
+        "{ polygonThreshold pt              |0.5| Confidence threshold of polygons. }"
+        "{ maxCandidate max                 |200| Max candidates of polygons. }"
+        "{ outputType type                  |0| The output type of detected text: 0: multi-oriented quadrilateral; 1: polygon }"
+        "{ unclipRatio ratio                |2.0| unclip ratio. }"
+        "{ vocabularyPath vp                | | Path to benchmarks for evaluation. "
+            "Download links are provided in doc/tutorials/dnn/dnn_text_spotting/dnn_text_spotting.markdown}";
```

Comment:
Is that possible to rewrite https://github.com/opencv/opencv/blob/master/samples/dnn/text_detection.cpp instead? This way we can proof that new API is a good replacement for old approach.

Reply:
The methods of scene text detection can be divided into (1) regression-based method and (2) segmentation-based method.
The original method (EAST) in text_detection.cpp is a regression-based method, while the new method is a segmentation-based method. Different post-processing is needed in the inference for them.

The methods of scene text recognition can be divided into (1) CTC-based method and (2) Attention-based method.
All of the samples adopts the CTC-based method (CRNN) for recognition because the current DNN module can not support some layers. In the following months, I will put some issues about them which are quite needed.

As you see, this is the first version of APIs for text detection and recognition. I will keep working on it and add more methods to make it as flexible as possible in the future.

Reply:
I just worry that we introduce an API which is not generic. Initial idea for the High Level API to help users start fast. Initially, we designed it for object detection networks which also have different post-processing depends on network family.

Reply:
I understand it, and I also have the aspiration to bring the latest topologies and algorithms to OpenCV.
As I discussed with the mentors before GSoC, we agreed on the current APIs to be the first version. I think it is a work in progress, and it is needed to be updated with more algorithms in the future.

Reply:
> we agreed on the current APIs to be the first version
we cannot merge something draft in `opencv/opencv`. API should be scalable and at least cover all the known topologies. There are multiple text detection and recognition network already supported by OpenCV (see the notes from https://github.com/opencv/opencv/blob/master/samples/dnn/text_detection.cpp). So introduced API should manage them.

Reply:
Ok, I will add those topologies to my API in this month. 
I have to prepare for a test this week.